scripts/doc_checker.py

```diff
@@ -24,97 +40,242 @@
+                    ),
                 )
+            )
+    return errors
+
+
+def check_spelling_words(doc: Docstring) -> list:
+    """
+    Check spelling of chosen words: "Modin", "pandas", "NumPy" in doc.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Docstring
+        Docstring handler.
+
+    Returns
+    -------
+    list
+        List of tuples with Modin error code and its description.
 
+    Notes
+    -----
+    "modin" word is treated as the constant.
+    """
+    if not doc.raw_doc:
+        return []
+    components = set(["Modin", "pandas", "NumPy"])
+    check_words = "|".join(x.lower() for x in components)
+
+    # comments work only with re.VERBOSE
+    pattern = r"""
+    (?:\W|^)                # non-capturing group: [^a-zA-Z0-9_] or start
+    ({check_words})         # words to check, example - "modin|pandas|numpy"
+    (?:                     # non-capturing group
+        [^"\.\/\w\\]        # any symbol except: '"', '.', '\' and any from [a-zA-Z0-9_]
+        | [\.]\s            # or '.' and any whitespace
+        | $                 # or end
+    )
+    """.format(
+        check_words=check_words
+    )
+    results = [
+        set(re.findall(pattern, line, re.I | re.VERBOSE)) - components
+        for line in doc.raw_doc.splitlines()
+    ]
+
+    errors = []
+    for line_idx, words_in_line in enumerate(results):
+        for word in words_in_line:
+            reference = [x for x in components if x.lower() == word.lower()][0]
+            errors.append(
+                (
+                    "MD02",
+                    MODIN_ERROR_CODES["MD02"].format(
+                        line=line_idx,
+                        word=word,
+                        reference=reference,
+                    ),
+                )
+            )
     return errors
 
 
-def update_results(results, modin_errors):
-    raise NotImplementedError
-"""
+def validate_modin_error(doc: Docstring) -> list:
+    """
+    Validate custom Modin errors.
+
+    Parameters
+    ----------
+    doc : numpydoc.validate.Docstring
+        Docstring handler.
+
+    Returns
+    -------
+    errors :
+        List of tuples with Modin error code and its description.
+    """
+    errors = check_optional_args(doc)
+    errors += check_spelling_words(doc)
+    return errors
 
 
-def skip_check_if_noqa(import_path):
+def update_results(results: dict, modin_errors: list):
     """
-    Align behavior with pydocstyle.
+    Add custom Modin errors to default numpydoc errors in results.
 
     Parameters
     ----------
-    import_path : str
-        Python-like import path.
+    results : dict
+        Dictionary that numpydoc.validate.validate return.
+    modin_errors : list
+        List of tuples with Modin error code and its description.
+
+    Returns
+    -------
+    dict
+        Updated dict with Modin custom errors.
+    """
+    for error in modin_errors:
+        results["errors"].append(error)
+    return results
```

Comment:
I'm failing to understand why you want to extract it into two separate functions... I'd merge `validate_modin_error()` and this function into a single one probably